src/command_buffer.rs

```diff
@@ -27,12 +27,13 @@ use crate::{align, DynamicBundle};
 /// assert!(world.contains(reserved_ent));
 /// ```
 pub struct CommandBuffer {
-    pub(crate) ent: Vec<EntityIndex>,
+    entity: Vec<EntityIndex>,
     storage: NonNull<u8>,
     layout: Layout,
     cursor: usize,
-    info: Vec<InfoBuffer>,
+    info: Vec<ComponentInfo>,
     ids: Vec<TypeId>,
+    mark: usize,
```

Comment:
This is very useful to indicate where we are in regards to a spawned entity, another solution would be to drain Vec as entities get spawned but I think that's not a very beneficial performance wise

If you have other solutions to this, please let me know 😃

Reply:
There's no reason for this information to be stored persistently inside the `CommandBuffer`; it is only needed inside `fn run_on`, and hence should be a local variable there, passed downward as needed (e.g. into a member of `ReadyBuffer`). Keeping it around in between calls is confusing, wastes storage, and increases the risk of bugs.